### PR TITLE
use restart_process extension

### PR DESCRIPTION
**Pull request by maiamcc**
Hello @nicks, @jazzdan,

Please review the following commits I made in branch maiamcc/use-restart-extension:

26e0f8eaa3314f6128c23f15f3b73b73e2a06375 (2020-05-15 19:04:08 -0400)
use restart_process extension

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by maiamcc** [comment]
Docs update here: https://github.com/tilt-dev/tilt.build/pull/585

File: 3-live-update/Tiltfile [hunk 1]
```diff
@@ -15,14 +20,15 @@ local_resource(
     resource_deps=['deploy'],
 )
 
-docker_build(
+docker_build_with_restart(
     'hello-tilt',
     'out',
+    entrypoint=['dotnet', 'hello-tilt.dll'],
```
Comment (nicks) [inline 3-live-update/Tiltfile:26]: hmmm...should we also update the built-in entrypoint parameter to accept an argv array? if so, is there an issue tracking that yet?

**Review by nicks — APPROVED** [review]
woooo

File: 3-live-update/Tiltfile (hunk 1 above)
Comment (maiamcc) [reply 3-live-update/Tiltfile:26]: Oh, it already does: https://github.com/tilt-dev/tilt/blob/ba1de77456a46ff72f5de798aaac092f47f481e7/internal/tiltfile/docker.go#L213 we just haven't updated the docs 😅

Comment (nicks) [reply 3-live-update/Tiltfile:26]: yay! :elephant: :elephant: :elephant:

**Pull request merged**